libsolidity/analysis/DeclarationTypeChecker.cpp

```diff
@@ -145,7 +145,11 @@ void DeclarationTypeChecker::endVisit(UserDefinedTypeName const& _typeName)
 	else if (EnumDefinition const* enumDef = dynamic_cast<EnumDefinition const*>(declaration))
 		_typeName.annotation().type = TypeProvider::enumType(*enumDef);
 	else if (ContractDefinition const* contract = dynamic_cast<ContractDefinition const*>(declaration))
+	{
+		if (contract->isLibrary())
+			m_errorReporter.typeError(1130_error, _typeName.location(), "Invalid use of a library name.");
```

Comment:
Since this is in the `DeclarationTypeChecker`, maybe "Invalid declaration using library.", or something similar to that that mentions "declaration"?

Reply:
``Invalid declaration using a library.`` sounds like there are valid declarations using a library...
``Libraries cannot be used in declarations.``... but then again you can declare a library in a "library declaration" after all :-D...
I'm not sure... that's the price of disallowing them all here - we need a generic error message.
``Libraries cannot occur in complex types.`` was an idea once, but this here now also catches ``L x;`` itself, which is not really a complex type...

Reply:
I could, however, probably make a simple ``L x;`` a special case as well, restoring the old error message for it and use ``Libraries cannot occur in complex types.`` for everything else...

Reply:
Nah... I'll just go for ``Invalid occurrence of a library name in a declaration.``.

Reply:
That sound good?

Reply:
Or no, hm...

Reply:
This will also catch stuff like ``function f() override (L)`` I think... which is really not a declaration position...
Maybe it's rather the ``DeclarationTypeChecker`` that is maybe named improperly :-D...

Reply:
Yea ok, then let's stay with the initial one